src/freenet/node/AnnounceSender.java

```diff
@@ -184,10 +184,8 @@ private void realRun() {
 				MessageFilter mfRejectedOverload = MessageFilter.create().setSource(next).setField(DMT.UID, uid).setTimeout(ACCEPTED_TIMEOUT).setType(DMT.FNPRejectedOverload);
 				MessageFilter mfOpennetDisabled = MessageFilter.create().setSource(next).setField(DMT.UID, uid).setTimeout(ACCEPTED_TIMEOUT).setType(DMT.FNPOpennetDisabled);
 
-				// mfRejectedOverload must be the last thing in the or
-				// So its or pointer remains null
-				// Otherwise we need to recreate it below
```

Comment:
Oof, this comment gives me goosebumps, and definitely _not_ in a good way… 😄

Have you made sure that whatever this comment is talking about is being handled correctly?

Reply:
I think this is a leftover from a time when this was spaghetti-code where they re-used the instances. That’s the only way how the comment could make sense.

We no longer do that, from what I can tell.